servo/logging.py

```diff
@@ -173,12 +173,16 @@ async def _process_queue(self) -> None:
             except asyncio.CancelledError:
                 raise
             except Exception as error:  # pylint: disable=broad-except
-                logger.warning(f"encountered exception while processing progress logging: {repr(error)}")
+                logger.warning(f"encountered exception while processing progress logging: {repr(progress)} => {repr(error)}")
                 if self._exception_handler:
-                    if asyncio.iscoroutinefunction(self._exception_handler):
-                        await self._exception_handler(error)
-                    else:
-                        self._exception_handler(error)
+                    try:
+                        if asyncio.iscoroutinefunction(self._exception_handler):
+                            await self._exception_handler(progress, error)
```

Comment:
Need to update the type annotation for the exception_handler argument here: https://github.com/opsani/servox/blob/fdfb83788e858fcf99e5e44f3b05f01a2eba3671/servo/logging.py#L81